lms/djangoapps/grades/management/commands/recalculate_subsection_grades.py

```diff
@@ -0,0 +1,98 @@
+"""
+Command to recalculate grades for all subsections with problem submissions
+in the specified time range.
+"""
+
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+from datetime import datetime
+import logging
+from pytz import utc
+
+from django.core.management.base import BaseCommand, CommandError
+from lms.djangoapps.grades.constants import ScoreDatabaseTableEnum
+from lms.djangoapps.grades.signals.handlers import PROBLEM_SUBMITTED_EVENT_TYPE
+from lms.djangoapps.grades.tasks import recalculate_subsection_grade_v3
+from courseware.models import StudentModule
+from student.models import user_by_anonymous_id
+from submissions.models import Submission
+from track.event_transaction_utils import create_new_event_transaction_id, set_event_transaction_type
+from util.date_utils import to_timestamp
+
+log = logging.getLogger(__name__)
+
+DATE_FORMAT = "%Y-%m-%d %H:%M"
+
+
+class Command(BaseCommand):
+    """
+    Example usage:
+        $ ./manage.py lms recalculate_subsection_grades
+            --modified_start '2016-08-23 16:43' --modified_end '2016-08-25 16:43' --settings=devstack
+    """
+    args = 'fill this in'
+    help = 'Recalculates subsection grades for all subsections modified within the given time range.'
+
+    def add_arguments(self, parser):
+        """
+        Entry point for subclassed commands to add custom arguments.
+        """
+        parser.add_argument(
+            '--dry_run',
+            action='store_true',
+            default=False,
+            dest='dry_run',
+            help="Output what we're going to do, but don't actually do it. To actually delete, use --delete instead."
+        )
+        parser.add_argument(
+            '--modified_start',
+            dest='modified_start',
+            help='Starting range for modified date (inclusive): e.g. "2016-08-23 16:43"; expected in UTC.',
+        )
+        parser.add_argument(
+            '--modified_end',
+            dest='modified_end',
+            help='Ending range for modified date (inclusive): e.g. "2016-12-23 16:43"; expected in UTC.',
+        )
+
+    def handle(self, *args, **options):
+        if 'modified_start' not in options:
+            raise CommandError('modified_start must be provided.')
+
+        if 'modified_end' not in options:
+            raise CommandError('modified_end must be provided.')
+
+        modified_start = utc.localize(datetime.strptime(options['modified_start'], DATE_FORMAT))
+        modified_end = utc.localize(datetime.strptime(options['modified_end'], DATE_FORMAT))
+        event_transaction_id = create_new_event_transaction_id()
+        set_event_transaction_type(u'lms.djangoapps.ARGBLARG')
```

Comment:
Also for some reason I can't reply to the other comment, but I'm just going to axe the option entirely. I don't handle --dry-run anywhere in the actual command and don't think we need to given the low potential of this command to cause any sort of "damage" if run with the wrong parameters.